src/main/java/de/hysky/skyblocker/skyblock/tabhud/screenbuilder/ScreenMaster.java

```diff
@@ -1,143 +1,217 @@
-                                .entrySet()) {
-
-                            try (BufferedReader reader = MinecraftClient.getInstance().getResourceManager()
-                                    .openAsReader(entry.getKey())) {
-                                JsonObject json = JsonParser.parseReader(reader).getAsJsonObject();
-                                if (json.get("format_version").getAsInt() != VERSION) {
-                                    throw new IllegalStateException(String.format("Resource pack isn't compatible! Expected version %d, got %d", VERSION, json.get("format_version").getAsInt()));
-                                }
-
-                            } catch (Exception ex) {
-                                throw new IllegalStateException(
-                                        "Rejected this resource pack. Reason: " + ex.getMessage());
-                            }
-                        }
-
-                        for (Map.Entry<Identifier, Resource> entry : manager
-                                .findResources("tabhud", path -> path.getPath().endsWith(".json") && !path.getPath().endsWith("version.json"))
-                                .entrySet()) {
-                            try {
-
-                                load(entry.getKey());
-                            } catch (Exception e) {
-                                LOGGER.error(e.getMessage());
-                                excnt++;
-                            }
-                        }
-                        if (excnt > 0) {
-                            throw new IllegalStateException("This screen definition isn't valid, see above");
-                        }
-                    }
-                });
+        });
+
+        ClientLifecycleEvents.CLIENT_STOPPING.register(client -> saveConfig());
+    }
+
+
+	/**
+	 * Filled at compile item with ASM.
+	 * Do not fill this method or change the signature unless you know what you're doing
+	 */
+	private static void instantiateWidgets() {}
+
+	/**
+	 * Called by the ASM generated code to add a widget instance to the map.
+	 * Do not change the signature unless you know what you're doing.
+	 */
+	public static void addWidgetInstance(HudWidget widget) {
+		HudWidget put = widgetInstances.put(widget.getInternalID(), widget);
+		if (widget instanceof TabHudWidget tabHudWidget) {
+			PlayerListMgr.tabWidgetInstances.put(tabHudWidget.getHypixelWidgetName(), tabHudWidget);
+		}
+		if (put != null) LOGGER.warn("[Skyblocker] Duplicate hud widget found: {}", widget);
+	}
+
+    /**
+     * @implNote !! The 3 first ones shouldn't be moved, ordinal is used in some places
+     */
+    public enum ScreenLayer {
+        MAIN_TAB,
+        SECONDARY_TAB,
+        HUD,
+        /**
+         * Default is only present for config and isn't used anywhere else
+         */
+        DEFAULT;
+
+        public static final Codec<ScreenLayer> CODEC = Codec.STRING.xmap(ScreenLayer::valueOf, ScreenLayer::name);
```

Comment:
If you're trying to make a `Codec` for this its preferable to use the `StringIdentifiable` interface